RA_Integration/IRA_Dialog.h

```diff
@@ -0,0 +1,100 @@
+#ifndef RA_DIALOG_H
+#define RA_DIALOG_H
+#pragma once
+
+#include <wtypes.h>
+
+
+/// <summary>
+/// Base class for all RA Dialogs.
+/// Not a "true" interface as not all dialogs share this functionality, but needed
+/// for the Message Queue.
+/// </summary>
+/// <remarks>
+///		Functions that are always different are pure virtual, 
+///		functions that are always the same are not virtual, 
+///		functions that are ususally the same but can be different are just virtual.
+/// </remarks>
+class IRA_Dialog
+{
+public:
+	// I can't figure out another way to do this
+	// basically just for those "four" dialogs in RA_Core you have to put false
+	IRA_Dialog(int resId, bool isModal = true/*, HWND parent = nullptr*/);
+	virtual INT_PTR DoModal(); // Most of the time doesn't need to be changed
+	// for the modeless, always the same for us so no overloads
+	// Unlike MFC this is not the handle to a parent window
+	virtual HWND Create();
+	static INT_PTR CALLBACK MsgQueue(HWND hwnd, UINT uMsg, WPARAM wParam, 
+		LPARAM lParam);
+
+	// Despite doing nothing this is necessary
```

Comment:
this is just a pure virtual function, no need to comment it.

Reply:
There is a need, it always returns 0, but we need a pure virtual for callbacks.

Reply:
I was just commenting on the comment. "Despite doing nothing this is necessary". The only thing that it seems to apply to is the prototype of a pure virtual function, which should be understood without the comment. If the comment applies to something else, please elaborate.

Reply:
It could be elaborated further, sure.